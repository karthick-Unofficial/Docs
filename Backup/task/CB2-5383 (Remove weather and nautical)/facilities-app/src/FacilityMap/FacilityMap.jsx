import React, { memo, useCallback, useState } from "react";
import PropTypes from "prop-types";
import ReactMapboxGl, { ZoomControl, RotationControl, Layer } from "react-mapbox-gl";
import NauticalChartsContainer from "./NauticalCharts/NauticalChartsContainer";
import RoadsAndLabelsContainer from "./RoadsAndLabels/RoadsAndLabelsContainer.js";
import WeatherRadarContainer from "./WeatherRadar/WeatherRadarContainer.js";
import MapControlsContainer from "./MapControls/MapControlsContainer.js";
import FacilitiesContainer from "./Facilities/FacilitiesContainer.js";
import CamerasContainer from "./Cameras/CamerasContainer";
import ActiveFloorPlanContainer from "./ActiveFloorPlan/ActiveFloorPlanContainer.js";
import { ContextMenuContainer } from "orion-components/Map/ContextMenu";
import AccessPointContainer from "./AccessPoint/AccessPointContainer";

const Map = ReactMapboxGl({
	accessToken:
		"pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA"
});

const setSpritePath = path =>
	path.replace(
		"please-replace-me-with-the-correct-hostname",
		location.hostname
	);


const propTypes = {
	style: PropTypes.string,
	setMapState: PropTypes.func.isRequired,
	zoom: PropTypes.number.isRequired,
	center: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.shape({
			lng: PropTypes.number.isRequired,
			lat: PropTypes.number.isRequired
		})
	]).isRequired,
	setMapReference: PropTypes.func.isRequired,
	toggleMapVisible: PropTypes.func.isRequired,
	baseMaps: PropTypes.array.isRequired
};

const defaultProps = {
	style: "satellite"
};

/**
 * TODO: This component could be added to orion-components as a BaseMap component (including the container, actions, zoom and rotation controls)
 * TODO: Wrap any child layers with theoretical BaseMap component
 */
const FacilityMap = ({
	style,
	setMapState,
	zoom,
	center,
	setMapReference,
	toggleMapVisible,
	baseMaps,
	nauticalChartsEnabled,
	weatherEnabled
}) => {
	const [mapRef, setMapRef] = useState(null);

	const getStyle = useCallback(() => {
		if (baseMaps.length > 0) {
			const baseMapSelection = baseMaps.filter((element) => element.name === style)[0];
			const spritePath = baseMapSelection.style;
			spritePath.sprite = setSpritePath(spritePath.sprite);
			return spritePath;
		}

	}, [baseMaps, style]);


	const handleMove = useCallback(
		map => {
			const mapCenter = Object.values(map.getCenter());
			const mapZoom = map.getZoom();
			setMapState({ mapCenter, mapZoom });
		},
		[setMapState]
	);
	const handleSetReference = map => {
		setMapReference(map);
		setMapRef(map);
		toggleMapVisible();
	};
	return (
		<Map
			style={getStyle()}
			center={center}
			maxZoom={18}
			minZoom={2}
			zoom={[zoom]}
			containerStyle={{
				width: "100vw",
				height: "calc(100vh - 48px)"
			}}
			movingMethod="flyTo"
			onMoveEnd={m => handleMove(m)}
			onStyleLoad={m => handleSetReference(m)}
		>
			{/* Layers for enforcing layer ordering */}
			<Layer id="---ac2-alerts-position-end" type="symbol" />
			<Layer id="---ac2-feed-entities-position-end" before="---ac2-alerts-position-end" type="symbol" />
			<Layer id="---ac2-weather-position-end" before="---ac2-feed-entities-position-end" type="symbol" />
			<Layer id="---ac2-floorplans-position-end" before="---ac2-weather-position-end" type="symbol" />
			<Layer id="---ac2-facilities-position-end" before="---ac2-floorplans-position-end" type="symbol" />
			<Layer id="---ac2-nautical-charts-position-end" before="---ac2-facilities-position-end" type="symbol" />

			<ZoomControl />
			<RotationControl />
			{!nauticalChartsEnabled ? null : <NauticalChartsContainer />}
			<RoadsAndLabelsContainer />
			{!weatherEnabled ? null : <WeatherRadarContainer />}
			<ActiveFloorPlanContainer />
			<FacilitiesContainer before="---ac2-facilities-position-end" />
			<CamerasContainer />
			<MapControlsContainer />
			{mapRef && (
				<ContextMenuContainer map={mapRef} />
			)}
			<AccessPointContainer />
		</Map>
	);
};

FacilityMap.propTypes = propTypes;
FacilityMap.defaultProps = defaultProps;

const onStyleChange = (prevProps, nextProps) => {
	if (prevProps.style !== nextProps.style) {
		return false;
	}
	if (prevProps.baseMaps !== nextProps.baseMaps) {
		return false;
	}
	return true;
};

export default memo(FacilityMap, onStyleChange);
