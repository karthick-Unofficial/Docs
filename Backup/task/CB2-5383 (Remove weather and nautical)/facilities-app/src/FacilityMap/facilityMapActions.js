import { updatePersistedState } from "orion-components/AppState/Actions";
export {
	setMapReference,
	toggleMapVisible
} from "orion-components/AppState/Actions";

export const setMapState = keyVal => {
	return dispatch => {
		dispatch(updatePersistedState("facilities-app", "mapSettings", keyVal));
	};
};
