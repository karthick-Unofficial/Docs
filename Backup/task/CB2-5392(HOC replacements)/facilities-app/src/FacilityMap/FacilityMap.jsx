import React, { memo, useCallback, useState, useEffect } from "react";
import PropTypes from "prop-types";
import ReactMapboxGl, { ZoomControl, RotationControl, Layer } from "react-mapbox-gl";
import NauticalCharts from "./NauticalCharts/NauticalCharts";
import RoadsAndLabels from "./RoadsAndLabels/RoadsAndLabels";
import WeatherRadar from "./WeatherRadar/WeatherRadar";
import MapControls from "./MapControls/MapControls";
import { FacilitiesLayer } from "orion-components/Map/Layers";
import * as facilitiesLayerActions from "./Facilities/facilitiesActions";
import Cameras from "./Cameras/Cameras";
import ActiveFloorPlan from "./ActiveFloorPlan/ActiveFloorPlan";
import { ContextMenu } from "orion-components/Map/ContextMenu";
import { AccessPointLayer } from "orion-components/Map/Layers";
import * as accessPointLayerAction from "./AccessPoint/accessPointActions";
import { setRTLTextPlugin } from "mapbox-gl";
import { useDispatch, useSelector } from "react-redux";
import {
	setMapReference,
	toggleMapVisible,
	setMapState
} from "./facilityMapActions";
import _ from "lodash";

const Map = ReactMapboxGl({
	accessToken:
		"pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA"
});

const setSpritePath = path =>
	path.replace(
		"please-replace-me-with-the-correct-hostname",
		location.hostname
	);


const propTypes = {
	style: PropTypes.string,
	setMapState: PropTypes.func.isRequired,
	zoom: PropTypes.number.isRequired,
	center: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.shape({
			lng: PropTypes.number.isRequired,
			lat: PropTypes.number.isRequired
		})
	]).isRequired,
	setMapReference: PropTypes.func.isRequired,
	toggleMapVisible: PropTypes.func.isRequired,
	baseMaps: PropTypes.array.isRequired,
	nauticalChartsEnabled: PropTypes.bool,
	weatherEnabled: PropTypes.bool
};

const defaultProps = {
	style: "satellite"
};

/**
 * TODO: This component could be added to orion-components as a BaseMap component (including the container, actions, zoom and rotation controls)
 * TODO: Wrap any child layers with theoretical BaseMap component
 */

function FacilityMapWrapper() {
	const clientConfig = useSelector(state => state.clientConfig);
	const baseMaps = useSelector(state => state.baseMaps);
	const Zoom = useSelector(state => state.clientConfig.mapSettings.zoom);
	const Center = useSelector(state => state.clientConfig.mapSettings.center);
	const { nauticalChartsEnabled, weatherEnabled } = _.size(clientConfig) && clientConfig.mapSettings;
	const mapZoom = useSelector(state => state.appState.persisted.mapSettings && state.appState.persisted.mapSettings.mapZoom);
	const mapCenter = useSelector(state => state.appState.persisted.mapSettings && state.appState.persisted.mapSettings.mapCenter);
	const mapStyle = useSelector(state => state.appState.persisted.mapSettings ? state.appState.persisted.mapSettings.mapStyle : "satellite");
	const zoom = mapZoom || Zoom;
	const center = mapCenter || Center
	const style = mapStyle;
	
	return <FacilityMap
		baseMaps={baseMaps}
		Center={Center}
		nauticalChartsEnabled={nauticalChartsEnabled}
		weatherEnabled={weatherEnabled}
		mapZoom={mapZoom}
		zoom={zoom}
		center={center}
		style={style}
	/>
}

const FacilityMap = memo((props) => {

	const { baseMaps, nauticalChartsEnabled, weatherEnabled, zoom, center, style } = props;
	const [mapRef, setMapRef] = useState(null);
	const dispatch = useDispatch();

	const getStyle = useCallback(() => {
		if (baseMaps.length > 0) {
			const baseMapSelection = baseMaps.filter((element) => element.name === style)[0];
			const spritePath = baseMapSelection.style;
			spritePath.sprite = setSpritePath(spritePath.sprite);
			return spritePath;
		}

	}, [baseMaps, style]);

	const handleMove = useCallback(
		map => {
			const mapCenter = Object.values(map.getCenter());
			const mapZoom = map.getZoom();
			dispatch(setMapState({ mapCenter, mapZoom }));
		},
		[setMapState]
	);
	const handleSetReference = map => {
		dispatch(setMapReference(map));
		setMapRef(map);
		dispatch(toggleMapVisible());
	};

	useEffect(() => {
		setRTLTextPlugin(
			"https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-rtl-text/v0.2.3/mapbox-gl-rtl-text.js",
			null,
			true
		);
	}, []);

	return (
		<Map
			style={getStyle()}
			center={center}
			maxZoom={18}
			minZoom={2}
			zoom={[zoom]}
			containerStyle={{
				width: "100vw",
				height: "calc(100vh - 48px)"
			}}
			movingMethod="flyTo"
			onMoveEnd={m => handleMove(m)}
			onStyleLoad={m => handleSetReference(m)}
		>
			{/* Layers for enforcing layer ordering */}
			<Layer id="---ac2-alerts-position-end" type="symbol" />
			<Layer id="---ac2-feed-entities-position-end" before="---ac2-alerts-position-end" type="symbol" />
			<Layer id="---ac2-weather-position-end" before="---ac2-feed-entities-position-end" type="symbol" />
			<Layer id="---ac2-floorplans-position-end" before="---ac2-weather-position-end" type="symbol" />
			<Layer id="---ac2-facilities-position-end" before="---ac2-floorplans-position-end" type="symbol" />
			<Layer id="---ac2-nautical-charts-position-end" before="---ac2-facilities-position-end" type="symbol" />

			<ZoomControl />
			<RotationControl />
			{!nauticalChartsEnabled ? null : <NauticalCharts />}
			<RoadsAndLabels />
			{!weatherEnabled ? null : <WeatherRadar />}
			<ActiveFloorPlan />
			<FacilitiesLayer before="---ac2-facilities-position-end" {...facilitiesLayerActions} />
			<Cameras />
			<MapControls />
			{mapRef && (
				<ContextMenu map={mapRef} />
			)}
			<AccessPointLayer labelsVisible={true} {...accessPointLayerAction}/>
		</Map>
	);
}, (prevProps, nextProps) => {
	if (prevProps.style !== nextProps.style) {
		return false;
	}
	if (prevProps.baseMaps !== nextProps.baseMaps) {
		return false;
	}
	return true;
});

FacilityMap.propTypes = propTypes;
FacilityMap.defaultProps = defaultProps;

export default FacilityMapWrapper;

