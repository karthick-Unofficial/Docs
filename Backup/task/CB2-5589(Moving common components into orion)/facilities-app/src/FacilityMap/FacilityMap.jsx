import React, { memo, useCallback, useState, useEffect } from "react";
import PropTypes from "prop-types";
import ReactMapboxGl, { ZoomControl, RotationControl, Layer } from "react-mapbox-gl";
import NauticalCharts from "./NauticalCharts/NauticalCharts";
import RoadsAndLabels from "./RoadsAndLabels/RoadsAndLabels";
import WeatherRadar from "./WeatherRadar/WeatherRadar";
import MapControls from "./MapControls/MapControls";
import { FacilitiesLayer, AccessPointLayer } from "orion-components/Map/Layers";
import * as facilitiesLayerActions from "./Facilities/facilitiesActions";
import Cameras from "./Cameras/Cameras";
import ActiveFloorPlan from "./ActiveFloorPlan/ActiveFloorPlan";
import { ContextMenu } from "orion-components/Map/ContextMenu";
import * as accessPointLayerAction from "./AccessPoint/accessPointActions";
import { setRTLTextPlugin, getRTLTextPluginStatus } from "mapbox-gl";
import { useDispatch, useSelector } from "react-redux";
import {
	setMapReference,
	toggleMapVisible,
	setMapState
} from "./facilityMapActions";
import size from "lodash/size";

const Map = ReactMapboxGl({
	accessToken:
		"pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA"
});

const setSpritePath = path =>
	path.replace(
		"please-replace-me-with-the-correct-hostname",
		location.hostname
	);

const propTypes = {
	style: PropTypes.string,
	zoom: PropTypes.number.isRequired,
	center: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.shape({
			lng: PropTypes.number.isRequired,
			lat: PropTypes.number.isRequired
		})
	]).isRequired,
	baseMaps: PropTypes.array.isRequired,
	nauticalChartsEnabled: PropTypes.bool,
	weatherEnabled: PropTypes.bool
};

const defaultProps = {
	style: "satellite"
};

/**
 * TODO: This component could be added to orion-components as a BaseMap component (including the container, actions, zoom and rotation controls)
 * TODO: Wrap any child layers with theoretical BaseMap component
 */

const FacilityMapWrapper = () => {
	const appState = useSelector(state => state.appState);
	const { persisted } = appState;
	const clientConfig = useSelector(state => state.clientConfig);
	const baseMaps = useSelector(state => state.baseMaps);
	const mapZoom = persisted.mapSettings && persisted.mapSettings.mapZoom;
	const mapCenter = persisted.mapSettings && persisted.mapSettings.mapCenter;
	const style = persisted.mapSettings ? persisted.mapSettings.mapStyle : "satellite";
	const zoom = persisted.mapSettings && mapZoom ? mapZoom : clientConfig.mapSettings.zoom;
	const center = persisted.mapSettings && mapCenter ? mapCenter : clientConfig.mapSettings.center;
	const maxZoom = clientConfig.mapSettings.maxZoom || 18;
	const minZoom = clientConfig.mapSettings.minZoom || 2;
	const { nauticalChartsEnabled, weatherEnabled } = size(clientConfig) && clientConfig.mapSettings;

	return <FacilityMap
		baseMaps={baseMaps}
		nauticalChartsEnabled={nauticalChartsEnabled}
		weatherEnabled={weatherEnabled}
		zoom={zoom}
		center={center}
		minZoom={minZoom}
		maxZoom={maxZoom}
		style={style}
	/>;
};

const FacilityMap = memo((props) => {
	const { baseMaps, nauticalChartsEnabled, weatherEnabled, zoom, center, minZoom, maxZoom, style } = props;
	const [mapRef, setMapRef] = useState(null);
	const dispatch = useDispatch();

	const getStyle = useCallback(() => {
		if (baseMaps.length > 0) {
			const baseMapSelection = baseMaps.filter((element) => element.name === style)[0];
			const spritePath = baseMapSelection.style;
			spritePath.sprite = setSpritePath(spritePath.sprite);
			return spritePath;
		}
	}, [baseMaps, style]);

	const handleMove = useCallback(
		map => {
			const mapCenter = Object.values(map.getCenter());
			const mapZoom = map.getZoom();
			setMapState({ mapCenter, mapZoom });
		},
		[setMapState]
	);

	const handleSetReference = map => {
		dispatch(setMapReference(map));
		setMapRef(map);
		dispatch(toggleMapVisible());
	};

	const setMinMaxZoom = (map) => {
		map.transform.minZoom = minZoom;
		map.transform.maxZoom = maxZoom;
	};

	useEffect(() => {
		if (getRTLTextPluginStatus() === "unavailable") {
			setRTLTextPlugin(
				"https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-rtl-text/v0.2.3/mapbox-gl-rtl-text.js",
				null,
				true
			);
		}
	}, []);

	return (
		<Map
			style={getStyle()}
			center={center}
			zoom={[zoom]}
			containerStyle={{
				width: "100vw",
				height: "calc(100vh - 48px)"
			}}
			movingMethod="flyTo"
			onMoveEnd={m => handleMove(m)}
			onStyleLoad={map => {
				handleSetReference(map);
				setMinMaxZoom(map);
			}}
		>
			{/* Layers for enforcing layer ordering */}
			<Layer id="---ac2-alerts-position-end" type="symbol" />
			<Layer id="---ac2-feed-entities-position-end" before="---ac2-alerts-position-end" type="symbol" />
			<Layer id="---ac2-weather-position-end" before="---ac2-feed-entities-position-end" type="symbol" />
			<Layer id="---ac2-floorplans-position-end" before="---ac2-weather-position-end" type="symbol" />
			<Layer id="---ac2-facilities-position-end" before="---ac2-floorplans-position-end" type="symbol" />
			<Layer id="---ac2-nautical-charts-position-end" before="---ac2-facilities-position-end" type="symbol" />

			<ZoomControl />
			<RotationControl />
			{!nauticalChartsEnabled ? null : <NauticalCharts />}
			<RoadsAndLabels />
			{!weatherEnabled ? null : <WeatherRadar />}
			<ActiveFloorPlan />
			<FacilitiesLayer before="---ac2-facilities-position-end" {...facilitiesLayerActions} />
			<Cameras />
			<MapControls />
			{mapRef && (
				<ContextMenu map={mapRef} />
			)}
			<AccessPointLayer labelsVisible={true} {...accessPointLayerAction} />
		</Map>
	);
}, (prevProps, nextProps) => {
	if (prevProps.style !== nextProps.style) {
		return false;
	}
	if (prevProps.baseMaps !== nextProps.baseMaps) {
		return false;
	}
	if (prevProps.center !== nextProps.center) {
		return false;
	}
	return true;
});

FacilityMap.propTypes = propTypes;
FacilityMap.defaultProps = defaultProps;
FacilityMap.displayName = "FacilityMap";

export default FacilityMapWrapper;