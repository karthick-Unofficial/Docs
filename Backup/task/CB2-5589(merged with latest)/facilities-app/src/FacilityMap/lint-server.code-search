# Query: lint-server
# ContextLines: 1

