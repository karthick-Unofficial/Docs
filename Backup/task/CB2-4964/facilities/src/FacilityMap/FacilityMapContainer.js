import { connect } from "react-redux";
import FacilityMap from "./FacilityMap.jsx";
import { bindActionCreators } from "redux";
import * as actionCreators from "./facilityMapActions";

const mapStateToProps = state => {
	const { appState, clientConfig, baseMaps } = state;
	const { persisted } = appState;
	const { zoom, center } = clientConfig.mapSettings;
	if (persisted.mapSettings) {
		const { mapZoom, mapCenter, mapStyle } = persisted.mapSettings;
		return {
			zoom: mapZoom || zoom,
			center: mapCenter || center,
			style: mapStyle,
			baseMaps
		};

	} else {
		return { zoom, center, baseMaps };
	}
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators(actionCreators, dispatch);
};

const FacilityMapContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(FacilityMap);

export default FacilityMapContainer;
