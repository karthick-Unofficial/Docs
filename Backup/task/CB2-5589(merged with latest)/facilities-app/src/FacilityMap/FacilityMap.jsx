import React, { memo, useCallback, useState, useEffect } from "react";
import PropTypes from "prop-types";
import ReactMapboxGl, { ZoomControl, RotationControl } from "react-mapbox-gl";
import MapControls from "./MapControls/MapControls";
import { FacilitiesLayer, AccessPointLayer, RoadsAndLabels, WeatherRadar, NauticalChartsLayer } from "orion-components/Map/Layers";
import * as facilitiesLayerActions from "./Facilities/facilitiesActions";
import Cameras from "./Cameras/Cameras";
import ActiveFloorPlan from "./ActiveFloorPlan/ActiveFloorPlan";
import { ContextMenu } from "orion-components/Map/ContextMenu";
import * as accessPointLayerAction from "./AccessPoint/accessPointActions";
import { setRTLTextPlugin, getRTLTextPluginStatus } from "mapbox-gl";
import { useDispatch, useSelector } from "react-redux";
import { setMapReference, toggleMapVisible, setMapState } from "./facilityMapActions";
import OrderingLayers from "orion-components/OrderingLayers/OrderingLayers";
import { CoordsOnCursor } from "orion-components/Map/Tools";

// cSpell:disable
const Map = ReactMapboxGl({
	accessToken: "pk.eyJ1IjoibGNsYXJrc29uIiwiYSI6ImNpaWh2dzlnaDAwZmp1ZGtueXdpa2loNXcifQ.gcbR_GT1AD2kUmIxkSxseA"
});
// cSpell:enable

const setSpritePath = (path) => path.replace("please-replace-me-with-the-correct-hostname", location.hostname);

const propTypes = {
	style: PropTypes.string,
	zoom: PropTypes.number.isRequired,
	center: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.shape({
			lng: PropTypes.number.isRequired,
			lat: PropTypes.number.isRequired
		})
	]).isRequired,
	baseMaps: PropTypes.array.isRequired,
	minZoom: PropTypes.number,
	maxZoom: PropTypes.number
};

const defaultProps = {
	style: "satellite"
};

/**
 * TODO: This component could be added to orion-components as a BaseMap component (including the container, actions, zoom and rotation controls)
 * TODO: Wrap any child layers with theoretical BaseMap component
 */

const FacilityMapWrapper = () => {
	const appState = useSelector((state) => state.appState);
	const { persisted } = appState;
	const clientConfig = useSelector((state) => state.clientConfig);
	const baseMaps = useSelector((state) => state.baseMaps);
	const mapZoom = persisted.mapSettings && persisted.mapSettings.mapZoom;
	const mapCenter = persisted.mapSettings && persisted.mapSettings.mapCenter;
	const style = persisted.mapSettings ? persisted.mapSettings.mapStyle : "satellite";
	const zoom = persisted.mapSettings && mapZoom ? mapZoom : clientConfig.mapSettings.zoom;
	const center = persisted.mapSettings && mapCenter ? mapCenter : clientConfig.mapSettings.center;
	const maxZoom = clientConfig.mapSettings.maxZoom || 18;
	const minZoom = clientConfig.mapSettings.minZoom || 2;

	return (
		<FacilityMap
			baseMaps={baseMaps}
			zoom={zoom}
			center={center}
			minZoom={minZoom}
			maxZoom={maxZoom}
			style={style}
		/>
	);
};

const FacilityMap = memo(
	(props) => {
		const { baseMaps, zoom, center, minZoom, maxZoom, style } = props;
		const [mapRef, setMapRef] = useState(null);
		const dispatch = useDispatch();

		const getStyle = useCallback(() => {
			if (baseMaps.length > 0) {
				const baseMapSelection = baseMaps.filter((element) => element.name === style)[0];
				const spritePath = baseMapSelection.style;
				spritePath.sprite = setSpritePath(spritePath.sprite);
				return spritePath;
			}
		}, [baseMaps, style]);

		const handleMove = useCallback((map) => {
			const mapCenter = Object.values(map.getCenter());
			const mapZoom = map.getZoom();
			setMapState({ mapCenter, mapZoom });
		}, []);

		const handleSetReference = (map) => {
			dispatch(setMapReference(map));
			setMapRef(map);
			dispatch(toggleMapVisible());
		};

		const setMinMaxZoom = (map) => {
			map.transform.minZoom = minZoom;
			map.transform.maxZoom = maxZoom;
		};

		useEffect(() => {
			if (getRTLTextPluginStatus() === "unavailable") {
				setRTLTextPlugin(
					"https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-rtl-text/v0.2.3/mapbox-gl-rtl-text.js",
					null,
					true
				);
			}
		}, []);

		return (
			<Map
				style={getStyle()}
				center={center}
				zoom={[zoom]}
				containerStyle={{
					width: "100vw",
					height: "calc(100vh - 48px)"
				}}
				movingMethod="flyTo"
				onMoveEnd={(m) => handleMove(m)}
				onStyleLoad={(map) => {
					handleSetReference(map);
					setMinMaxZoom(map);
				}}
			>
				<OrderingLayers />

				<ZoomControl />
				<RotationControl />
				<NauticalChartsLayer />
				{mapRef && <RoadsAndLabels map={mapRef} />}
				<WeatherRadar weatherReload={false} />
				<ActiveFloorPlan />
				<FacilitiesLayer before="---ac2-facilities-position-end" {...facilitiesLayerActions} />
				<Cameras />
				<MapControls />
				{mapRef && <ContextMenu map={mapRef} />}
				<AccessPointLayer
					labelsVisible={true}
					{...accessPointLayerAction}
					before="---ac2-feed-entities-position-end"
				/>
				<CoordsOnCursor />
			</Map>
		);
	},
	(prevProps, nextProps) => {
		if (prevProps.style !== nextProps.style) {
			return false;
		}
		if (prevProps.baseMaps !== nextProps.baseMaps) {
			return false;
		}
		if (prevProps.center !== nextProps.center) {
			return false;
		}
		return true;
	}
);

FacilityMap.propTypes = propTypes;
FacilityMap.defaultProps = defaultProps;
FacilityMap.displayName = "FacilityMap";

export default FacilityMapWrapper;
