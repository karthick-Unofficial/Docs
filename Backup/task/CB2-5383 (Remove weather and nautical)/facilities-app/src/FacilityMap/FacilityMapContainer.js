import { connect } from "react-redux";
import FacilityMap from "./FacilityMap.jsx";
import { bindActionCreators } from "redux";
import * as actionCreators from "./facilityMapActions";

const mapStateToProps = state => {
	const { appState, clientConfig, baseMaps } = state;
	const { persisted } = appState;
	const { zoom, center } = clientConfig.mapSettings;
	const { nauticalChartsEnabled, weatherEnabled } = _.size(clientConfig) && clientConfig.mapSettings;

	if (persisted.mapSettings) {
		const { mapZoom, mapCenter, mapStyle } = persisted.mapSettings;
		return {
			zoom: mapZoom || zoom,
			center: mapCenter || center,
			style: mapStyle,
			baseMaps,
			nauticalChartsEnabled,
			weatherEnabled
		};

	} else {
		return { zoom, center, baseMaps, nauticalChartsEnabled, weatherEnabled };
	}
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators(actionCreators, dispatch);
};

const FacilityMapContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(FacilityMap);

export default FacilityMapContainer;
