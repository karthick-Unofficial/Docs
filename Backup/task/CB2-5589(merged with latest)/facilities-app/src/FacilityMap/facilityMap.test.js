import React from "react";
import FacilityMap from "./FacilityMap";
import { shallow } from "enzyme";
import { useSelector } from "react-redux";

jest.mock("mapbox-gl/dist/mapbox-gl", () => ({
	Map: () => ({})
}));

jest.mock("react-redux", () => ({
	useSelector: jest.fn()
}));

describe("FacilityMap", () => {
	it("renders", () => {
		useSelector
			.mockImplementationOnce(() => {
				return { persisted: { mapSettings: { zoom: 5 } } };
			})
			.mockImplementationOnce(() => {
				return { mapSettings: { zoom: 5 } };
			});

		const wrapper = shallow(<FacilityMap />);
		expect(wrapper).toMatchSnapshot();
	});
});
