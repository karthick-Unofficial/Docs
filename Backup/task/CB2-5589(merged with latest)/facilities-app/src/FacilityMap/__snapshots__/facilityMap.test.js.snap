// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityMap renders 1`] = `ShallowWrapper {}`;
